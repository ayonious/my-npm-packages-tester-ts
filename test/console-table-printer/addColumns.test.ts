import { expect } from '@jest/globals';
import { Table } from 'console-table-printer';

describe('Table with addColumns and addColumn', () => {
  it('should add a single column using addColumn', () => {
    const table = new Table();
    
    table.addColumn({ 
      name: 'id', 
      alignment: 'left',
      color: 'blue'
    });

    table.addRow({ id: 1 });
    table.addRow({ id: 2 });

    expect(table.render()).toMatchSnapshot();
  });

  it('should add multiple columns using addColumns', () => {
    const table = new Table();
    
    table.addColumns([
      { name: 'id', alignment: 'left' },
      { name: 'name', alignment: 'center' },
      { name: 'price', alignment: 'right', color: 'green' }
    ]);

    table.addRow({ id: 1, name: 'Product A', price: 99.99 });
    table.addRow({ id: 2, name: 'Product B', price: 149.99 });

    expect(table.render()).toMatchSnapshot();
  });

  it('should combine addColumn and addColumns', () => {
    const table = new Table();
    
    // Add first column individually
    table.addColumn({ 
      name: 'id', 
      alignment: 'left' 
    });

    // Add multiple columns at once
    table.addColumns([
      { name: 'name', alignment: 'center' },
      { name: 'category', alignment: 'center' }
    ]);

    // Add another single column
    table.addColumn({ 
      name: 'price', 
      alignment: 'right',
      color: 'yellow'
    });

    table.addRow({ id: 1, name: 'Product A', category: 'Electronics', price: 99.99 });
    table.addRow({ id: 2, name: 'Product B', category: 'Books', price: 149.99 });

    expect(table.render()).toMatchSnapshot();
  });
}); 