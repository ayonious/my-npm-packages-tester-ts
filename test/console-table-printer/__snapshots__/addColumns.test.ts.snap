// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Table with addColumns and addColumn should add a single column using addColumn 1`] = `
"┌────┐
│[37m [0m[01mid[0m │
├────┤
│[37m [0m[34m1 [0m │
│[37m [0m[34m2 [0m │
└────┘"
`;

exports[`Table with addColumns and addColumn should add multiple columns using addColumns 1`] = `
"┌────┬───────────┬────────┐
│[37m [0m[01mid[0m │[37m [0m[01m  name   [0m │[37m [0m[01m price[0m │
├────┼───────────┼────────┤
│[37m [0m[37m1 [0m │[37m [0m[37mProduct A[0m │[37m [0m[32m 99.99[0m │
│[37m [0m[37m2 [0m │[37m [0m[37mProduct B[0m │[37m [0m[32m149.99[0m │
└────┴───────────┴────────┘"
`;

exports[`Table with addColumns and addColumn should combine addColumn and addColumns 1`] = `
"┌────┬───────────┬─────────────┬────────┐
│[37m [0m[01mid[0m │[37m [0m[01m  name   [0m │[37m [0m[01m category  [0m │[37m [0m[01m price[0m │
├────┼───────────┼─────────────┼────────┤
│[37m [0m[37m1 [0m │[37m [0m[37mProduct A[0m │[37m [0m[37mElectronics[0m │[37m [0m[33m 99.99[0m │
│[37m [0m[37m2 [0m │[37m [0m[37mProduct B[0m │[37m [0m[37m   Books   [0m │[37m [0m[33m149.99[0m │
└────┴───────────┴─────────────┴────────┘"
`;
